### PR TITLE
Fixes #28219 - setting 201 as response code for POST

### PR DESCRIPTION
**Pull request by sseelam2**

### CODE REVIEW

**Comment by jturel** [comment]
[test katello]

**Comment by chris1984** [comment]
Starting to review

**Comment by jturel** [comment]
The change requests that the response code returns 201 instead of 200 so it looks like this might not be completely working - definitely needs a test case!

**Comment by chris1984** [comment]
> The change requests that the response code returns 201 instead of 200 so it looks like this might not be completely working - definitely needs a test case!

Good catch, I read it wrong. Will remove my review

**Comment by jturel** [comment]
@sseelam2 check the redmine issue linked on the BZ. The PR associated with it references a method called `respond_for_create` maybe we can use that? This solution makes me thing we'll respond with a 201 anywhere that we don't set a response code ourselves. That doesn't seem correct: 

```
The HTTP 201 Created success status response code indicates that the request has succeeded and has led to the creation of a resource. The new resource is effectively created before this response is sent back and the new resource is returned in the body of the message, its location being either the URL of the request, or the content of the Location header.

The common use case of this status code is as the result of a POST request.
```

Let me know if I have misinterpreted your change

**Comment by sseelam2** [comment]
> The change requests that the response code returns 201 instead of 200 so it looks like this might not be completely working - definitely needs a test case!

@jturel Only for a POST request right? So maybe I could set the response to 201 inside respond_for_create if it gets successfully created.

**Comment by jturel** [comment]
> > The change requests that the response code returns 201 instead of 200 so it looks like this might not be completely working - definitely needs a test case!
> 
> @jturel Only for a POST request right? So maybe I could set the response to 201 inside respond_for_create if it gets successfully created.

Correct, since we use POST for creating resources in the REST world. Try it and see. I recommend starting with a failing test case before making any change :)

**Comment by sseelam2** [comment]
[test katello]

File: test/controllers/api/v2/host_collections_controller_test.rb [hunk 1]
```diff
@@ -61,7 +61,7 @@ def test_show
     def test_create
       post :create, params: { :organization_id => @organization, :host_collection => {:name => 'Collection A', :description => 'Collection A, World Cup 2014'} }
 
-      assert_response :success
+      assert_response (201)
```
Comment (jturel) [inline test/controllers/api/v2/host_collections_controller_test.rb]: Let's be consistent with our style. I'd like to see `assert_response 201` or `assert_response(201)` (note no space when using parenthesis)

**Comment by jturel** [comment]
I think the commit & PR need a more accurate name since the change isn't actually to set 201 response code for POST requests, but for create requests (they just happen to be HTTP POST).

File: test/controllers/api/v2/host_collections_controller_test.rb (hunk 1 above)
Comment (sseelam2) [reply test/controllers/api/v2/host_collections_controller_test.rb]: rubocop complained about it as well.

**Comment by jturel** [comment]
LGTM!

@chris1984 wanna give it another test?

**Comment by chris1984** [comment]
Sounds good will test again and report back

**Pull request merged**